apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/DcGeoRDALoadJob.java

```diff
@@ -0,0 +1,149 @@
+  public static final int SCAN_INTERVAL_DEFAULT = 300;
+  public static final int RUN_TIME_DEFAULT = 300;
+  public static final int MAX_RECORDS_DEFAULT = Integer.MAX_VALUE;
+  public static final String BATCH_SIZE_PROPERTY = "DCGeoBatchSize";
+  public static final int BATCH_SIZE_DEFAULT = 1;
+  public static final String CALLS_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "calls");
+  public static final String FAILURES_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "failures");
+  public static final String SUCCESSES_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "successes");
+  public static final String PROCESSED_METER_NAME =
+      MetricRegistry.name(DcGeoRDALoadJob.class.getSimpleName(), "processed");
+
+  private final Config config;
+  private final Callable<RDASource<PreAdjudicatedClaim>> sourceFactory;
+  private final Callable<RDASink<PreAdjudicatedClaim>> sinkFactory;
+  private final Meter callsMeter;
+  private final Meter failuresMeter;
+  private final Meter successesMeter;
+  private final Meter processedMeter;
+
+  public DcGeoRDALoadJob(
+      Config config,
+      Callable<RDASource<PreAdjudicatedClaim>> sourceFactory,
+      Callable<RDASink<PreAdjudicatedClaim>> sinkFactory,
+      MetricRegistry appMetrics) {
+    this.config = config;
+    this.sourceFactory = sourceFactory;
+    this.sinkFactory = sinkFactory;
+    callsMeter = appMetrics.meter(CALLS_METER_NAME);
+    failuresMeter = appMetrics.meter(FAILURES_METER_NAME);
+    successesMeter = appMetrics.meter(SUCCESSES_METER_NAME);
+    processedMeter = appMetrics.meter(PROCESSED_METER_NAME);
+  }
+
+  /**
+   * Factory method to construct a new job instance using standard parameters.
+   *
+   * @param appMetrics MetricRegistry used to track operational metrics
+   * @return a DcGeoRDALoadJob instance suitable for use by PipelineManager.
+   */
+  public static DcGeoRDALoadJob newDcGeoRDALoadJob(MetricRegistry appMetrics) {
+    return new DcGeoRDALoadJob(
+        new Config(),
+        () -> new SkeletonRDASource(appMetrics),
+        () -> new SkeletonRDASink(appMetrics),
+        appMetrics);
+  }
+
+  public Duration getScanInterval() {
+    return config.getScanInterval();
+  }
+
+  @Override
+  public PipelineJobOutcome call() throws Exception {
+    final long startMillis = System.currentTimeMillis();
+    int processedCount = 0;
+    Exception error = null;
+    try {
+      callsMeter.mark();
+      try (RDASource<PreAdjudicatedClaim> source = sourceFactory.call();
+          RDASink<PreAdjudicatedClaim> sink = sinkFactory.call()) {
+        processedCount =
+            source.retrieveAndProcessObjects(
+                config.getMaxObjectsPerCall(), config.getBatchSize(), config.getMaxRunTime(), sink);
+      }
+    } catch (ProcessingException ex) {
+      processedCount += ex.getProcessedCount();
+      error = ex.getCause();
+    } catch (Exception ex) {
```

Comment:
I imagine you know this, given your review of the orchestration PR, but just a heads-up that this will need to be tweaked once that's merged, to allow for interrupts.